src/lib/yang/data.lua

```diff
@@ -298,18 +306,18 @@ local function struct_parser(keyword, members, ctype)
          -- Scalar/array parser responsible for requiring whitespace
          -- after keyword.  Struct/table don't need it as they have
          -- braces.
-         local sub = assert(members[k], 'unrecognized parameter: '..k)
+         local sub = assert(expanded_members[k], 'unrecognized parameter: '..k)
          local id = normalize_id(k)
-         ret[id] = sub.parse(P, ret[id])
+         ret[id] = sub.parse(k, P, ret[id])
          P:skip_whitespace()
       end
-      for _,k in ipairs(keys) do
+      for k,_ in pairs(expanded_members) do
          local id = normalize_id(k)
-         ret[id] = members[k].finish(ret[id])
+         ret[id] = expanded_members[k].finish(ret[id], k)
       end
       return ret
    end
-   local function parse(P, out)
+   local function parse(k, P, out)
```

Comment:
I agree, I probably should have it as the last parameter.

k is actually the leaf name but maybe it stands for keyword. It's not the same keyword as `keyword` though, confusingly. Hopefully this'll be clearer with the above change as it doesn't even get used in this function. It's really only used in the choice handler's parser.